engines/bops_core/app/components/bops_core/table_of_contents_component.rb

```diff
@@ -2,8 +2,41 @@
 
 class BopsCore::TableOfContentsComponent < ViewComponent::Base
   def initialize(items:)
-    @items = items
+    @items = build_items(items)
   end
 
   attr_reader :items
+
+  private
+
+  def build_items(items)
+    Array(items).filter_map do |section|
+      links = visible_links(section[:links])
+      next unless links.any?
```

Comment:
Minor point but because this is `filter_map` not `each`, you could return `nil` here instead of using `next` (because nils will be excluded). I'm not sure if it makes a difference though.